d/d.bzl

```diff
@@ -152,60 +152,61 @@ def _setup_deps(deps, name, working_dir):
+    d_srcs = []
     transitive_d_srcs = []
     versions = []
-    imports = depset()
-    link_flags = depset()
-    symlinked_libs = depset()
+    imports = []
+    link_flags = []
+    symlinked_libs = []
     for dep in deps:
         if hasattr(dep, "d_lib"):
             # The dependency is a d_library.
-            libs = depset([dep.d_lib], transitive = [libs])
+            libs.append(dep.d_lib)
             transitive_libs.append(dep.transitive_libs)
-            symlinked_libs = depset([dep.d_lib], transitive = [dep.transitive_libs, symlinked_libs])
-            d_srcs = depset(dep.d_srcs, transitive = [d_srcs])
+            symlinked_libs.append(depset([dep.d_lib], transitive = [dep.transitive_libs]))
+            d_srcs += dep.d_srcs
             transitive_d_srcs.append(dep.transitive_d_srcs)
             versions += dep.versions + ["Have_%s" % _format_version(dep.label.name)]
-            link_flags = depset(["-L-l%s" % dep.label.name] + dep.link_flags, transitive = [link_flags])
-            imports = depset(["%s/%s" % (dep.label.package, im) for im in dep.imports], transitive = [imports])
+            link_flags += ["-L-l%s" % dep.label.name] + dep.link_flags
+            imports += ["%s/%s" % (dep.label.package, im) for im in dep.imports]
 
         elif hasattr(dep, "d_srcs"):
             # The dependency is a d_source_library.
-            d_srcs = depset(dep.d_srcs, transitive = [d_srcs])
+            d_srcs += dep.d_srcs
             transitive_d_srcs.append(dep.transitive_d_srcs)
             transitive_libs.append(dep.transitive_libs)
-            symlinked_libs = depset(transitive = [dep.transitive_libs, symlinked_libs])
-            link_flags = depset(["-L%s" % linkopt for linkopt in dep.linkopts], transitive = [link_flags])
-            imports = depset(["%s/%s" % (dep.label.package, im) for im in dep.imports], transitive = [imports])
+            symlinked_libs.append(dep.transitive_libs)
+            link_flags += ["-L%s" % linkopt for linkopt in dep.linkopts]
+            imports += ["%s/%s" % (dep.label.package, im) for im in dep.imports]
             versions += dep.versions
 
         elif CcInfo in dep:
             # The dependency is a cc_library
             native_libs = a_filetype(_get_libs_for_static_executable(dep))
-            libs = depset(native_libs, transitive = [libs])
+            libs.extend(native_libs)
             transitive_libs.append(depset(native_libs))
-            symlinked_libs = depset(native_libs, transitive = [symlinked_libs])
-            link_flags = depset(["-L-l%s" % dep.label.name], transitive = [link_flags])
+            symlinked_libs.append(depset(native_libs))
+            link_flags += ["-L-l%s" % dep.label.name]
 
         else:
             fail("D targets can only depend on d_library, d_source_library, or " +
                  "cc_library targets.", "deps")
 
+    symlinked_libs = depset(transitive = symlinked_libs)
     for symlinked_libs in symlinked_libs.to_list():
```

Comment:
These two lines are confusing, you first convert a list to a depset using the same variable name, then iterate over it with the same variable name again.

Reply:
Thanks! I didn't notice the loop variable was the same. That's super confusing.

I've renamed the variable, inlined the call to depset, and used a list comprehension.